js/src/components/shipping-rate-section/minimum-order-card/minimum-order-form-modals/add-minimum-order-form-modal.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import MinimumOrderFormModal from './minimum-order-form-modal';
+
+/**
+ * @typedef { import(".~/data/actions").CountryCode } CountryCode
+ * @typedef { import("../typedefs.js").MinimumOrderGroup } MinimumOrderGroup
+ */
+
+/**
+ * Display the add minimum order modal that is wrapped in a Form.
+ *
+ * When users submit the form, `props.onRequestClose` will be called first, and then followed by `props.onSubmit`.
+ * If we were to call `props.onSubmit` first, it may cause some state change in the parent component and causing this component not to be rendered,
+ * and when `props.onRequestClose` is called later, there would be a runtime React error because the component is no longer there.
+ *
+ * @param {Object} props Props.
+ * @param {Array<CountryCode>} props.countryOptions Array of country codes options, to be used as options in SupportedCountrySelect.
+ * @param {MinimumOrderGroup} props.initialValues Initial values for the form.
+ * @param {(values: MinimumOrderGroup) => void} props.onSubmit Callback when the form is submitted, with the form value.
+ * @param {() => void} props.onRequestClose Callback to close the modal.
+ */
+const AddMinimumOrderFormModal = ( {
+	countryOptions,
+	initialValues,
+	onSubmit,
+	onRequestClose = noop,
```

Comment:
As in https://github.com/woocommerce/google-listings-and-ads/pull/1424#discussion_r860287002 I gout cough with https://github.com/woocommerce/google-listings-and-ads/issues/1439

:nail_care: YAGNI, we don't need a default.

Reply:
Let's have a quick discussion on this. It is a practice that I learned from past experience, and if it is not a good practice, I should really change it from now. 😄 

The reason I put in `noop` or `() => {}` as default value for this functions is to prevent runtime error and prevent blowing up the whole page. When someone missed to specify the function and it breaks, it might just break one small part of the UI, and users may still be able to continue with their main flow without being impacted.

This is the closest article I found on the internet related to this topic - https://www.digitalocean.com/community/tutorials/how-to-customize-react-components-with-props:

> It’s good practice to always add a default value to prevent runtime errors if a prop is not required.

Curious to hear your thoughts on this. 🙏

Reply:
Yep, blowing up the whole page is definitely not something we want. And I also hate when something crashes in our plugin, and we end up with a blank page of doom.

But to me, this seems to be a systematic/platform problem (that's another reason I like native Web Platform - if your custom element throws, everything around works intact).

In the article, you linked there is: "**If a prop isn’t required**" and I think in our app those callbacks are required.

I see this as a hard trade:
In case of a bug having an explicit, easy to debug error, with a risk to blow UX vs having just impaired but hopefully semi-functional UI but silent, hard to track bug.


Also, we don't set defaults for other props, why? We do not protect our component against not providing or providing wrong values for other args. Like, what if `countryOptions` is malformed/missing?

----

Speaking of the practice itself, I think it's a good one if applied to reusable published components. When we do not know/control how they will be used. But when this "someone" who may miss is also us - we are the authors and consumers - we control everything. Then preparing for scenarios that we know, are not there is a wasted effort.

To protect against failures, personally, I'd add more tests and make sure the platform we use can isolate crashes gracefully as Web does.

So I'm fine with keeping it here for sake of future usage. I just thought :nail_care: it may never happen that we would really have such scenario, and it's not worth anticipating everything.

Reply:
> In the article, you linked there is: "If a prop isn’t required" and I think in our app those callbacks are required.

Yeah. If there is a linting that will throw error saying we didn't provide a value for required props, that will be really good. (I used to have that in my past experience, I think that was a React + TypeScript project.)

> Speaking of the practice itself, I think it's a good one if applied to _reusable published components_. _When we do not know/control how they will be used_. But when this "someone" who may miss is also us - we are the authors and consumers - we control everything. Then preparing for scenarios that we know, are not there is a wasted effort.

Ahh, I think this is a very good point, and I think now I know why or how I got that practice from. 🙂 Thanks. I'll proceed to remove the `noop`. 

> To protect against failures, personally, I'd add more tests and make sure the platform we use can isolate crashes gracefully as Web does.

I was just thinking to say adding the defensive `noop` code can be complementary to adding more tests, both can exist together.

Reply:
I have removed the `noop` usage in [`e452bed` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/e452bedf64da09c9f73a9a77de7ae3e7077f5188). 🙂